config/configauth/pdatacontext.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type ctxKey struct{}
+
+type AuthContext interface {
+	Equal(other interface{}) bool
+	GetAttribute(attrName string) interface{}
+}
+
+func InjectAuthContext(ctx context.Context, attrs map[interface{}]interface{}) context.Context {
```

Comment:
Seems like we are forced to pass `map[interface{}]interface{}` here because w use `authC` in this package. I think we can pass `AuthContext` as a parameter and `authC` becomes an implementation detail of specific auth extension (we can make `authC` a helper if we think it is a common implementation for many auth extensions).
Removing `authC` from this package increases decoupling between `AuthContext` interface and specific implementations, which I think is beneficial.
What do you think?